src/AzureClient/Visualization/TargetStatusEncoders.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Azure.Quantum.Client.Models;
+using Microsoft.Jupyter.Core;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    internal static class TargetStatusExtensions
+    {
+        internal static Dictionary<string, object> ToDictionary(this TargetStatus target) =>
+            new Dictionary<string, object>()
+            {
+                { "targetName", target.Id },
+                { "currentAvailability", target.CurrentAvailability },
+                { "averageQueueTime", target.AverageQueueTime },
+                { "statusPage", target.StatusPage },
+            };
+
+        internal static Table<TargetStatus> ToJupyterTable(this IEnumerable<TargetStatus> targets) =>
+            new Table<TargetStatus>
+            {
+                Columns = new List<(string, Func<TargetStatus, string>)>
+                {
+                    ("Target Name", target => target.Id),
+                    ("Current Availability", target => target.CurrentAvailability),
+                    ("Average Queue Time", target => target.AverageQueueTime.ToString()),
+                    ("Status Page", target => target.StatusPage),
+                },
+                Rows = targets.ToList()
+            };
+    }
+
+    public class TargetStatusToHtmlEncoder : IResultEncoder
+    {
+        private static readonly IResultEncoder tableEncoder = new TableToHtmlDisplayEncoder();
+
+        public string MimeType => MimeTypes.Html;
+
+        public EncodedData? Encode(object displayable)
+        {
+            if (displayable is TargetStatus target) displayable = new List<TargetStatus>() { target };
```

Comment:
Very minor, but given that you've used this pattern a few times so far, would it be worth a new extension method that ensures something is an `IEnumerable`?

```
internal static IEnumerable<T>? AsEnumerableOf<T>(this object? source) =>
    source match
    {
        T singleton => new List<T> { singleton },
        IEnumerable<T> collection => collection.
        _ => null
    };
```
My thought with the naming here matches was to match the `as` operator by returning `null` if the type comparison fails.

Reply:
Yes, that's a good suggestion, I'll do this.